src/test/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImplTest.java

```diff
@@ -1,658 +1,590 @@
+import uk.gov.hmcts.reform.professionalapi.domain.Organisation;
+import uk.gov.hmcts.reform.professionalapi.domain.OrganisationStatus;
+import uk.gov.hmcts.reform.professionalapi.domain.PrdEnum;
+import uk.gov.hmcts.reform.professionalapi.domain.PrdEnumId;
+import uk.gov.hmcts.reform.professionalapi.domain.ProfessionalUser;
+import uk.gov.hmcts.reform.professionalapi.domain.RoleAdditionResponse;
+import uk.gov.hmcts.reform.professionalapi.domain.RoleDeletionResponse;
+import uk.gov.hmcts.reform.professionalapi.domain.RoleName;
+import uk.gov.hmcts.reform.professionalapi.domain.SuperUser;
+import uk.gov.hmcts.reform.professionalapi.domain.UserProfile;
+import uk.gov.hmcts.reform.professionalapi.domain.UserProfileUpdatedData;
+import uk.gov.hmcts.reform.professionalapi.persistence.OrganisationRepository;
+import uk.gov.hmcts.reform.professionalapi.persistence.PrdEnumRepository;
+import uk.gov.hmcts.reform.professionalapi.persistence.ProfessionalUserRepository;
+import uk.gov.hmcts.reform.professionalapi.persistence.UserAttributeRepository;
+
+public class ProfessionalUserServiceImplTest {
+
+    private final ProfessionalUserRepository professionalUserRepository = Mockito.mock(ProfessionalUserRepository.class);
+    private final OrganisationRepository organisationRepository = mock(OrganisationRepository.class);
+    private final UserAttributeRepository userAttributeRepository = mock(UserAttributeRepository.class);
+    private final PrdEnumRepository prdEnumRepository = mock(PrdEnumRepository.class);
+    private final UserProfileFeignClient userProfileFeignClient = mock(UserProfileFeignClient.class);
+    private final UserAttributeServiceImpl userAttributeService = mock(UserAttributeServiceImpl.class);
+    private final FeignException feignExceptionMock = mock(FeignException.class);
+
+    private final Organisation organisation = new Organisation("some-org-name", null, "PENDING", null, null, null);
+    private final UserProfile userProfile = new UserProfile(UUID.randomUUID().toString(), "test@email.com", "fName", "lName", IdamStatus.PENDING);
+    private final GetUserProfileResponse getUserProfileResponseMock = new GetUserProfileResponse(userProfile, false);
+
+    private final ProfessionalUserServiceImpl professionalUserService = new ProfessionalUserServiceImpl(
+            organisationRepository, professionalUserRepository, userAttributeRepository,
+            prdEnumRepository, userAttributeService, userProfileFeignClient);
+
+    private final ProfessionalUser professionalUser = new ProfessionalUser("some-fname", "some-lname", "some-email", organisation);
+
+    private final SuperUser superUser = new SuperUser("some-fname", "some-lname", "some-super-email", organisation);
+
+    private List<PrdEnum> prdEnums = new ArrayList<>();
+    private List<String> userRoles = new ArrayList<>();
+
+    @Before
+    public void setup() {
+        userRoles.add("pui-user-manager");
+        PrdEnumId prdEnumId = mock(PrdEnumId.class);
```

Comment:
you can use actual object for PrdEnumId

Reply:
I have addressed this in the following commit: https://github.com/hmcts/rd-professional-api/pull/450/commits/8bbca5f0c3802efae79bc99bc40a954ba2d528d2